File: java/leetcode_grind/Day899.java
package leetcode_grind;

public class Day899 {
    // https://leetcode.com/problems/build-array-from-permutation/description/?envType=daily-question&envId=2025-05-06
    static class Solution1 {
        public int[] buildArray(int[] nums) {
            int n = nums.length;       
            for (int i = 0; i < n; i++) {
                nums[i] += 1000 * (nums[nums[i]] % 1000);
            }
            for (int i = 0; i < n; i++) {
                nums[i] /= 1000;
            }
            return nums;
        }
    }
}

File: src/leetcode_grind/day_899.rs
// https://leetcode.com/problems/build-array-from-permutation/description/?envType=daily-question&envId=2025-05-06
pub fn build_array(mut nums: Vec<i32>) -> Vec<i32> {
    let n = nums.len();        
    for i in 0..n {
        nums[i] += 1000 * (nums[nums[i] as usize] % 1000);
    }
    for i in 0..n {
        nums[i] /= 1000;
    }
    nums
}